drivers/can/can.c

```diff
@@ -407,46 +406,48 @@ static int can_open(FAR struct file *filep)
    * for this device, then perform hardware initialization.
    */
 
-  if (list_is_empty(&dev->cd_readers))
+  caninfo("ocount: %u\n", dev->cd_crefs);
+
+  if (dev->cd_crefs >= 255)
     {
-      caninfo("ocount: %d\n", 0);
+      /* Limit to no more than 255 opens */
```

Comment:
LGTM; we were missing this before and could rollover if too many opens.

Reply:
It was tracked before my changes, but was relying on the readers list. Now since the readers list is not updated in case of the write only we can't use the readers list any more

Reply:
Ah yes I see that now. Still, this change is an improvement because it is becoming more consistent with the way other drivers are written.